### ISSUE TITLE
make cleanthirdparty broken on the emulator

### ISSUE THREAD

**Issue by tigran123** [koreader/kindlepdfviewer#357]
With the recent changes to Makefile we now have to type this to do cleanthirdparty even on emulator:

```
make CHOST=arm-kindle-linux-gnueabi cleanthirdparty
```

because if we type the usual `make cleanthirdparty` then we'll get error from luajit's Makefile:

```
$ make cleanthirdparty
make -C luajit-2.0 CC="gcc" HOST_CC="gcc -m32" CROSS="arm-none-linux-gnueabi-" clean
make[1]: Entering directory `/home/tigran/Kindle/kindlepdfviewer/luajit-2.0'
make -C src clean
make[2]: arm-none-linux-gnueabi-gcc: Command not found
make[2]: Entering directory `/home/tigran/Kindle/kindlepdfviewer/luajit-2.0/src'
make[2]: arm-none-linux-gnueabi-gcc: Command not found
make[2]: arm-none-linux-gnueabi-gcc: Command not found
make[2]: arm-none-linux-gnueabi-gcc: Command not found
make[2]: arm-none-linux-gnueabi-gcc: Command not found
make[2]: arm-none-linux-gnueabi-gcc: Command not found
Makefile:251: *** Unsupported target architecture.  Stop.
make[2]: Leaving directory `/home/tigran/Kindle/kindlepdfviewer/luajit-2.0/src'
make[1]: *** [clean] Error 2
make[1]: Leaving directory `/home/tigran/Kindle/kindlepdfviewer/luajit-2.0'
make: *** [cleanthirdparty] Error 2
```

**Comment by NiLuJe** [comment]
The cross TC isn't in your PATH?

I can fallback to the previous ugly fix of simply unsetting CFLAGS, but I thought this was slightly less ugly ;).

(The issue being if you happen to have CFLAGS set in your env with a -march set, all hell breaks loose).

I'll look into it, see if I can come up with something that doesn't require the cross-tc on clean ;).

**Comment by tigran123** [comment]
Of course it is in my PATH. Otherwise how I would have been compiling KPV for the past three months :)
But cross TC should be _irrelevant_  for `make cleanthirdparty` in the emulator mode.

**Comment by tigran123** [comment]
Btw, we need clear instructions on upgrading TC to the one that works with your recent changes, because x-tools no longer does, see issue #359 I have just opened.

**Comment by NiLuJe** [comment]
Have a few errands to run, will fix when I get back.

basically,

make -C luajit-2.0 CC=$(HOSTCC) CFLAGS=$(BASE_CFLAGS) clean

should do the trick (and fix both issues).

**Comment by tigran123** [comment]
Oh, it will fix #359 as well? Cool, I will test it, thank you! (though I can't imagine how the change to luajit's compile fixes PDF opening problem, but I believe you!)

**Comment by tigran123** [comment]
Yes, cleanthirdparty is fixed by the above (and ignore my reference to #359, I didn't notice "clean" at the end, I thought you are changing the actual luajit compile)

**Comment by NiLuJe** [comment]
Great!

I'll push a PR with this (among other things) after testing a few things, hopefully fixing this, #341 & #359

**Comment by tigran123** [comment]
Ok, thanks. My 2012.03 TC is still downloading... I am hoping that using it will fix #359.
